lib/_http_client.js

```diff
@@ -620,6 +620,10 @@ function responseOnEnd() {
   const res = this;
   const req = this.req;
 
+  if (req.socket) {
+    req.socket.removeListener('timeout', emitRequestTimeout);
```

Comment:
This is now run unconditionally even if `listenSocketTimeout` was never called right?

Reply:
yes

Reply:
could probably add a check for `req.timeoutCb`?

Reply:
Was thinking the same.